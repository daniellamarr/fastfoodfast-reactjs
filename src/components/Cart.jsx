/* eslint-disable no-plusplus */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getCartItems } from '../actions/cartActions';

/**
 * Cart
 */
export class Cart extends Component {
  /**
   * Initialized functions for component
   * @returns {null} null
   */
  componentDidMount() {
    this.props.getCartItems();
  }

  /**
   * @returns {JSX} jsx
   */
  render() {
    return (
      <div>
        <ul>
        {
          this.props.cart.cart.map((item, index) => (
            <li key={index}>
              <a href={`menu/${item.id}`}>
                <img src={item.image} alt=""/>
                <p className="cart-title">
                  {item.title}
                </p>
                <p>
                  {`${item.quantity} x ${item.price}`}
                </p>
              </a>
            </li>
          ))
        }
        </ul>
      </div>
    );
  }
}

Cart.propTypes = {
  getCartItems: PropTypes.func.isRequired,
  cart: PropTypes.shape({
    cart: PropTypes.arrayOf(PropTypes.shape({
      title: PropTypes.string.isRequired,
    })),
    noOfItems: PropTypes.number,
  }),
};

export const mapStateToProps = state => ({
  cart: state.cart,
});

export default connect(mapStateToProps, { getCartItems })(Cart);
