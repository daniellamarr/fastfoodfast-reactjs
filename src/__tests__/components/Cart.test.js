/* eslint-disable no-undef */
import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { Cart } from '../../components/Cart.jsx';

const middlewares = [thunk];

const mockStore = configureMockStore(middlewares);
const store = mockStore();

let wrapper;

const props = {
  getCartItems: () => {},
  cart: {
    cart: [
      {
        title: 'Hello Test',
      },
    ],
    noOfItems: 1,
  },
};

beforeEach(() => {
  wrapper = mount(
    <Provider store={store}>
      <Cart {...props} />
    </Provider>,
  );
});
afterEach(() => {
  wrapper.unmount();
});

describe('SingleMenu Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
