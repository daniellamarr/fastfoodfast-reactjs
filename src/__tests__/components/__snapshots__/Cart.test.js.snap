// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleMenu Component should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Cart
    cart={
      Object {
        "cart": Array [
          Object {
            "title": "Hello Test",
          },
        ],
        "noOfItems": 1,
      }
    }
    getCartItems={[Function]}
  >
    <div>
      <ul>
        <li>
          <a
            href="menu/undefined"
          >
            <img
              alt=""
            />
            <p
              className="cart-title"
            >
              Hello Test
            </p>
            <p>
              undefined x undefined
            </p>
          </a>
        </li>
      </ul>
    </div>
  </Cart>
</Provider>
`;
